# import all library

In [5]:
from pathlib import Path
import pandas as pd

# import function

In [6]:
from function.ttest_function import categorical_vs_numerical

# load data

In [7]:
df = pd.read_csv('../../00-dataSource/dataSourceV2/clean_data_v1.csv')
df.head()

,Age,Gender,Daily_Screen_Time(hrs),Sleep_Quality(1-10),Stress_Level(1-10),Days_Without_Social_Media,Exercise_Frequency(week),Social_Media_Platform,Happiness_Index(1-10)
0,44,male,3.1,7.0,6.0,2.0,5.0,facebook,10.0
1,30,other,5.1,7.0,8.0,5.0,3.0,linkedin,10.0
2,23,other,7.4,6.0,7.0,1.0,3.0,youtube,6.0
3,36,female,5.7,7.0,8.0,1.0,1.0,tiktok,8.0
4,34,female,7.0,4.0,7.0,5.0,1.0,x (twitter),8.0


## hypothesis between gender & stress level

In [8]:
ttest_results = pd.DataFrame([
    categorical_vs_numerical(df, 'Gender', 'Stress_Level(1-10)'),
])

ttest_results

,Test,Category Variable,Numerical Variable,Number of Groups,Group Means,Statistic,p-value,Decision,Conclusion
0,One-way ANOVA,Gender,Stress_Level(1-10),3,"{'female': 6.6026, 'male': 6.6331, 'other': 6....",0.0235,0.976754,Fail to reject H0,The difference in Stress_Level(1-10) across Ge...


The ANOVA test results show that gender does not have a statistically significant effect on stress level (p = 0.9768). 

The mean stress levels across gender groups are nearly identical, indicating that gender does not play an important role in influencing stress in this dataset. 

This suggests that stress is more likely influenced by behavioral factors such as screen time and sleep quality rather than demographic variables like gender.